src/node_api.cc

```diff
@@ -89,8 +103,8 @@ class BufferFinalizer : private Finalizer {
             env,
             finalizer->_finalize_data,
             finalizer->_finalize_hint);
-      });
-    });
+       });
+    }
```

Comment:
Can you store the lambda above in a variable and either pass it to `SetImmediate` or call it directly, instead of duplicating the code?

Reply:
sure, just pushed a an update which looks like it resolved the crashes on master. Will look at refactoring once I confirm that it fixes the crashses we were seeing on master.

Reply:
The code is not quite the same anymore, but there is still some that should be able to be shared.